stellargraph/layer/gcn.py

```diff
@@ -149,8 +188,20 @@ def get_config(self):
 
 class GCN:
     """
-    To create GCN layers with Keras layers.
-
+    A stack of Graph Convolutional layers to implement the graph convolution network model as in https://arxiv.org/abs/1609.02907
+
+    The model minimally requires specification of the layer sizes as a list of ints
+    corresponding to the feature dimensions for each hidden layer,
+    activation functions for each hidden layers, and a generator object.
+
+    Args:
+        layer_sizes (list of int): list of output sizes of GCN layers in the stack
+        activations (list of str): list of activations applied to each layer's output
+        generator (FullBatchNodeGenerator): an instance of FullBatchNodeGenerator class constructed on the graph of interest
+        bias (bool): toggles an optional bias in GCN layers
+        dropout (float): dropout rate applied to input features of each GCN layer
+        kernel_regularizer (str): normalization applied to the kernels of GCN layers
+        kwargs: additional parameters for chebyshev or localpool filters
```

Comment:
is there a description of what these additional parameters are, and how to use them? If not, it should be provided.

Reply:
Thanks, added.

Reply:
You've given an example of `filter="localpool", max_degree=2` as additional arguments to the `GCN` class constructor, but in `gcn-cora-example.py` the argument `filter="localpool"` is passed to the `FullBatchNodeGenerator`, not to `GCN`! This is quite confusing and in fact misleading. Please fix properly.

Reply:
GCN class constructor also has `filter="localpool"` in the `gcn-cora-example.py` in the VERY right in the L61 because L61 is quite long and is hidded by the browser.  please see https://github.com/stellargraph/stellargraph/blob/hotfix/gcn-doc/demos/node-classification-gcn/gcn-cora-example.py#L61.

Reply:
ok, thanks, didn't notice. This however raises other questions, e.g., are the `filter` arguments passed to `FullBatchNodeGenerator` and to `GCN` required to be the same? What if they are not? Can the `filter` type be inferred from the `generator` passed to `GCN`, rather than be specified as an optional argument? Think of a "fool proof" way of making this work.

Reply:
Also, if `filter` argument can be passed to `FullBatchNodeGenerator` as a kwarg, a description should also be added in file:///Users/tys017/Projects/Graph_Analytics/code/stellar-ml/docs/_build/html/api.html#stellargraph.mapper.FullBatchNodeGenerator

Reply:
Updated